share/tilt_wii.c

```diff
@@ -184,13 +184,12 @@ void tilt_free(void)
         state.status = 0;
         SDL_mutexV(mutex);
 
-        /* Kill the thread and destroy the mutex. */
+        /* Destroy the mutex and terminate the thread. */
 
-        SDL_KillThread(thread);
         SDL_DestroyMutex(mutex);
-
         mutex  = NULL;
         thread = NULL;
+        SDL_WaitThread(thread, NULL);
```

Comment:
This waits on a NULL thread. The waiting should happen before the call to SDL_DestroyMutex, where SDL_KillThread used to occur.